src/dune_project.ml

```diff
@@ -173,6 +173,21 @@ module Source_kind = struct
       ]
 end
 
+module File_key = struct
+  type t = string
+
+  module Map = String.Map
+
+  let of_string s = s
+  let to_string s = s
+
+  let make ~name ~root =
+    let root = Path.Local.to_string (Path.Source.to_local root) in
+    let name = Name.to_dyn name in
```

Comment:
Why these conversions? `Digest.generic` can take anything so we might as well pass it the raw values. It would only be an issue if the values if we had maps, but this is not the case here.

Reply:
In case we ever try to intern the paths again. We don't want to be interning integers.

Reply:
I guess. However, we don't do that in other places and it's not ideal to have inconsistencies like this in the code base. If we intern paths again, we should just go over all calls to `Digest.generic` and decide what to do. If we have half of the calls written this way and half of the calls just digesting the raw value, it's not very clear what we are doing.

In conclusion, we should just digest `(name, root)` directly here.

Reply:
ok, done.